### PR TITLE
Update to egui/eframe 0.20.0

### PR DESCRIPTION
**Pull request by Wumpf**
had to run cargo update to resolve a dependency issue when building with wgpu, keeping those changes here

### CODE REVIEW

**Comment by Wumpf** [comment]
this only works with trunk egui ofc

File: src/main.rs [hunk 1]
```diff
@@ -17,7 +17,8 @@ fn main() {
 
 // when compiling to web using trunk.
 #[cfg(target_arch = "wasm32")]
-fn main() {
+#[wasm_bindgen::prelude::wasm_bindgen(start)]
+pub async fn main_js() {
```
Comment (emilk) [inline src/main.rs:21]: If we move this to `lib.rs`, we should be able to remove the `fn main()` below

**Comment by coderedart** [comment]
one solution is adding `wasm-bindgen-futures` as dependency. 

then just renaming current main fn to `fake_main`
```rust
// this is the new main. use wasm_biindgen_futures to spawn it
#[cfg(target_arch = "wasm32")]
fn main() {
    wasm_bindgen_futures::spawn_local(fake_main());
}
// this is the previous main fn. just add async to it
#[cfg(target_arch = "wasm32")]
pub async fn fake_main() {
}
```

**Comment by Wumpf** [comment]
`pollster::block_on` breaks down when using WebGPU as a backend as this actually needs to hand back to the browser during device creation.

**Comment by emilk** [comment]
@coderedart btw, `trunk serve` recompiles all dependencies when changing anything in `main.rs`. This is… very annoying

**Comment by emilk** [comment]
`wasm_bindgen_futures::spawn_local` works well, and seems less hacky!

**Comment by coderedart** [comment]
> pollster::block_on breaks down when using WebGPU as a backend as this actually needs to hand back to the browser during device creation.
ooh, good to know.  thanks. I completely forgot that webgpu is a thing.

> @coderedart btw, `trunk serve` recompiles all dependencies when changing anything in `main.rs`. This is… very annoying

I am relieved that its not just me. I just assumed it was rust-analyzer noise because when i pass `--release` flag, it doesn't recompile deps, but without the flag, it will recompile some dependencies. I will dig deeper today

**Comment by coderedart** [comment]
@emilk yeah, its rust analyzer clashing with cargo. even without trunk, after a change, running `cargo build --target=wasm32-unknown-unknown` will trigger full recompilation. if i just edit using vim without RA, then the dependencies are not recompiled.

**Pull request merged**